txp/Pos/Txp/Core/Tx.hs

```diff
@@ -66,7 +70,7 @@ topsortTxs toTx input =
             tsVisited %= HS.insert txHash
             let visitedNew = HS.insert txHash visitedThis
                 dependsUnfiltered =
-                    nub $ mapMaybe (\x -> HM.lookup (txInHash x) txHashes)
+                    nub $ mapMaybe txByInput
```

Comment:
@volhovM please review `topsortTx` changes (maybe you're going to review full PR, I dunno).